Test interdependence issue

Whilst changing the existing `grunt coverage-integration` task to be `grunt coverage-all` in #6704 , I initially put the unit tests after the integration etc tests. That resulted in two tests inside of `unit/config_spec.js` failing with the output below.

Putting the unit tests first resolves the issue enough for #6704, but it would be good to resolve this issue as it means the tests are affecting each other and they definitely shouldn't be.

```
1738 passing (2m)
  11 pending
  2 failing

  1) Config Theme should have exactly the right keys:
     AssertionError: expected Object {
  cover: 'casper',
  description: 'casper',
  logo: 'casper',
  navigation: Array [ Object { label: 'Home', url: '/' } ],
  permalinks: '/:year/:month/:day/:slug/',
  postsPerPage: '5',
  title: 'casper',
  url: 'http://my-ghost-blog.com'
} to have keys url, title, description, logo, cover
	extra keys: navigation, postsPerPage, permalinks
      at Assertion.fail (node_modules/should/lib/assertion.js:91:17)
      at Assertion.Object.defineProperty.value (node_modules/should/lib/assertion.js:163:19)
      at Context.<anonymous> (core/test/unit/config_spec.js:41:37)

  2) Config Url urlFor should return url for an author when asked for:

      AssertionError: expected '/author/ghost-owner/' to be '/author/joe-bloggs/'
      + expected - actual

      -/author/ghost-owner/
      +/author/joe-bloggs/
      
      at Assertion.fail (node_modules/should/lib/assertion.js:91:17)
      at Assertion.Object.defineProperty.value (node_modules/should/lib/assertion.js:163:19)
      at Context.<anonymous> (core/test/unit/config_spec.js:299:61)
```

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -230,10 +230,10 @@ var _              = require('lodash'),
                 },
                 coverage_all: {
                     src: [
-                        'core/test/unit',
-                        'core/server/apps',
                         'core/test/integration',
-                        'core/test/functional'
+                        'core/server/apps',
+                        'core/test/functional',
+                        'core/test/unit'
                     ],
                     options: {
                         coverageFolder: 'core/test/coverage/all',

--- a/core/test/functional/routes/channel_spec.js
+++ b/core/test/functional/routes/channel_spec.js
@@ -232,6 +232,8 @@ describe('Channel Routes', function () {
             }).catch(done);
         });
 
+        after(testUtils.teardown);
+
         it('should 404 for /tag/ route', function (done) {
             request.get('/tag/')
                 .expect('Cache-Control', testUtils.cacheRules.private)

--- a/core/test/unit/config_spec.js
+++ b/core/test/unit/config_spec.js
@@ -17,6 +17,10 @@ var should         = require('should'),
 i18n.init();
 
 describe('Config', function () {
+    before(function () {
+        configUtils.restore();
+    });
+
     afterEach(function () {
         configUtils.restore();
     });

--- a/core/test/utils/fixtures/data-generator.js
+++ b/core/test/utils/fixtures/data-generator.js
@@ -245,7 +245,9 @@ DataGenerator.forKnex = (function () {
         clients;
 
     function createBasic(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             uuid: uuid.v4(),
             created_by: 1,
             created_at: new Date(),
@@ -255,7 +257,9 @@ DataGenerator.forKnex = (function () {
     }
 
     function createPost(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             uuid: uuid.v4(),
             status: 'published',
             html: overrides.markdown,
@@ -291,7 +295,9 @@ DataGenerator.forKnex = (function () {
     }
 
     function createUser(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             uuid: uuid.v4(),
             status: 'active',
             created_by: 1,
@@ -316,7 +322,9 @@ DataGenerator.forKnex = (function () {
     }
 
     function createAppField(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             uuid: uuid.v4(),
             created_by: 1,
             created_at: new Date(),
@@ -328,7 +336,9 @@ DataGenerator.forKnex = (function () {
     }
 
     function createAppSetting(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             uuid: uuid.v4(),
             app_id: 1,
             created_by: 1,
@@ -337,7 +347,9 @@ DataGenerator.forKnex = (function () {
     }
 
     function createToken(overrides) {
-        return _.defaults(overrides, {
+        var newObj = _.cloneDeep(overrides);
+
+        return _.defaults(newObj, {
             token: uuid.v4(),
             client_id: 1,
             expires: Date.now() + globalUtils.ONE_DAY_MS

